client/services-payment/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+//! RPC client for Services Payment pallet
+
+pub use pallet_services_payment_runtime_api::ServicesPaymentApi as ServicesPaymentRuntimeApi;
+use {
+    core::marker::PhantomData,
+    jsonrpsee::{
+        core::{async_trait, RpcResult},
+        proc_macros::rpc,
+    },
+    sc_client_api::UsageProvider,
+    sp_api::ProvideRuntimeApi,
+    sp_runtime::traits::Block as BlockT,
+    std::sync::Arc,
+};
+
+#[rpc(server)]
+pub trait ServicesPaymentApi<Balance, ParaId> {
+    #[method(name = "tanssi_servicesPaymentBlockCost")]
+    async fn block_cost(&self, para_id: ParaId) -> RpcResult<Balance>;
+
+    #[method(name = "tanssi_servicesPaymentCollatorAssignmentCost")]
+    async fn collator_assignment_cost(&self, para_id: ParaId) -> RpcResult<Balance>;
```

Comment:
I'm not sure if we want those methods to have a para id as input, the current implementation of `ProvideBlockProductionCost` and `ProvideCollatorAssignmentCost` ignores the para id, because the cost is the same for all para ids.

But on the other hand, if we later change the cost to depend on the para id, we will need to change the rpc method as well...

Reply:
I had the same thoughts, added it as I saw that the the Providers do receive the ParaId but ignores it. Is it planned for the cost to change depending on the parachain?

Reply:
I don't know, it would be a good idea to have a different cost if the para id is a parathread or a parachain. But in that case it would be enough to have a bool as input, not a para id.

And the reason we have the para id param in the provider trait is because it was added a long time ago, not because we are planning to do something with it.

I'm thinking that let's just leave it with a para id param, and before the release we revisit this, to avoid breaking the rpc in the future.

Reply:
I see, sounds good. I've created a ticket to revisit this.